testing/scripts/wait-for-file

```diff
@@ -3,21 +3,22 @@
 # Sleeps until a file comes into existence.
 
 if [[ $# -ne 2 ]]; then
-    echo >&2 "usage: $0 <file to wait for> <max secs to wait>"
+    echo >&2 "usage: $0 <file to wait for> <max cycles to wait>"
```

Comment:
I'd leave this at `max secs` to sleep (it's easier to reason about).

The maybe something like:
```
sleep_until=$(date "-s +$max_wait sec" +%s)
...
if [[ $(date +%s) -gt $sleep_until ]] ;then
  break;
fi
```
Your `bc` approach also looked fine, was there something off with it?

This maybe does the wait_count's well enough?
```
$ echo 'x=(0.0 / 0.1 + 0.5); scale=0; if (x/1 > 0) { x/1 } else { 1 }' | bc
```

Reply:
The issue with the `date` comparison is a bit that if the time goes into the past by a few years, it'll sleep very long, and it involves many more `date` invocations.


This would be okay, too, I guess:
```
sleep_interval_ms=100
wait_count = $(( ${max_wait}000 / ${sleep_interval_ms} ))
```

Reply:
I actually kind of prefer something like this, though it's slightly weird. I'll push that and see what you/CI think